core/src/main/java/nu/yona/server/exceptions/ResourceBasedException.java

```diff
@@ -1,21 +1,44 @@
 package nu.yona.server.exceptions;
 
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.context.event.EventListener;
 import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+import nu.yona.server.Translator;
 
 /**
  * This exception class is used as a base for all exceptions that use an error message which is defined in the message properties
- * 
- * @author pgussow
  */
 public abstract class ResourceBasedException extends RuntimeException
 {
+	@Component
+	private static class TranslatedTextRetriever
+	{
+		@Autowired
+		private Translator translator;
+
+		@EventListener({ ContextRefreshedEvent.class })

```

Comment:
Hmm, maybe the current structure is better because then `Translator` does not need to know about `ResourceBasedException`.
I have a name suggestion though: `TranslatorInjector`, it could set a static `Translator` object.